src/js/app.js

```diff
@@ -132,14 +133,17 @@ const Application = App.extend({
     ];
   },
 
-  onFail(options, response) {
-    addError(response);
-    this.getRegion('preloader').show(new PreloaderView({ notSetup: true }));
+  onFail(options, error = {}) {
+    const { response, responseData } = error;
+
+    addError(responseData || error);
```

Comment:
This `onFail` might handle an api error or a code error.  But most errors by this point will be handled generically.